src/plugins/template/src/template_compiled_model.cpp

```diff
@@ -0,0 +1,247 @@
+    try {
+        compile_model(m_model);
+    } catch (const InferenceEngine::Exception&) {
+        throw;
+    } catch (const std::exception& e) {
+        IE_THROW(Unexpected) << "Standard exception from compilation library: " << e.what();
+    } catch (...) {
+        IE_THROW(Unexpected) << "Generic exception is thrown";
+    }
+}
+// ! [executable_network:ctor_cnnnetwork]
+
+// ! [executable_network:map_graph]
+// forward declaration
+void transform_model(const std::shared_ptr<ov::Model>& model);
+
+void TemplatePlugin::CompiledModel::compile_model(const std::shared_ptr<ov::Model>& model) {
+    // apply plugins transformations
+    transform_model(model);
+    // Generate backend specific blob mappings. For example Inference Engine uses not ngraph::Result nodes friendly name
+    // as inference request output names but the name of the layer before.
+    // TODO: Remove it
+    size_t idx = 0;
+    for (auto&& result : model->get_results()) {
+        const auto& input = result->input_value(0);
+        auto name = ov::op::util::get_ie_output_name(input);
+        if (_outputIndex.emplace(name, idx).second)
+            idx++;
+    }
+    for (auto&& parameter : model->get_parameters()) {
+        _inputIndex.emplace(parameter->get_friendly_name(), model->get_parameter_index(parameter));
+    }
+
+    // Perform any other steps like allocation and filling backend specific memory handles and so on
+}
+// ! [executable_network:map_graph]
+
+// ! [executable_network:create_infer_request]
+std::shared_ptr<InferenceEngine::IInferRequestInternal> TemplatePlugin::CompiledModel::create_infer_request() const {
+    auto internal_request = create_sync_infer_request();
+    return std::make_shared<TemplateAsyncInferRequest>(
+        std::static_pointer_cast<TemplatePlugin::TemplateInferRequest>(internal_request),
+        get_task_executor(),
+        get_template_plugin()->_waitExecutor,
+        get_callback_executor());
+}
+
+std::shared_ptr<InferenceEngine::IInferRequestInternal> TemplatePlugin::CompiledModel::create_sync_infer_request()
+    const {
+    std::vector<std::shared_ptr<const ov::Node>> _inputs, _outputs;
+    for (const auto& output : m_model->inputs()) {
+        _inputs.emplace_back(output.get_node_shared_ptr());
+    }
+    for (const auto& output : outputs()) {
+        _outputs.emplace_back(output.get_node_shared_ptr());
+    }
+
+    return std::make_shared<TemplateInferRequest>(
+        _inputs,
+        _outputs,
+        std::static_pointer_cast<const TemplatePlugin::CompiledModel>(shared_from_this()));
+}
+// ! [executable_network:create_infer_request]
+
+void TemplatePlugin::CompiledModel::set_property(const ov::AnyMap& properties) {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+ov::RemoteContext TemplatePlugin::CompiledModel::get_context() const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::shared_ptr<ov::Model> TemplatePlugin::CompiledModel::get_runtime_model() const {
+    return m_model->clone();
```

Comment:
why do we need to clone?

Reply:
We need to do it because user can change runtime model and it will affect our internal model

Reply:
Changed API and removed